course_discovery/apps/course_metadata/search_indexes.py

```diff
@@ -410,6 +418,37 @@ def prepare_search_card_display(self, obj):
 
             return []
         return [degree.search_card_ranking, degree.search_card_cost, degree.search_card_courses]
+    
+    def prepare_num_of_courses(self, obj):
+        return obj.courses.count()
+    
+    def prepare_banner_image_url(self, obj):
+        if obj.banner_image_url:
+            return obj.banner_image_url
+        elif obj.banner_image:
+            return obj.banner_image.url
```

Comment:
what if we don't have `obj.banner_image.url`, Will the code break?

Reply:
The function will return None in that case (not explicitly mentioned), this won't break anything as the field is nullable.